reference/debugging.md

```diff
@@ -0,0 +1,229 @@
+# Debugging
+
+If you’re experiencing issues with the Ambassador Edge Stack and cannot diagnose the issue through the "Diagnostics" tab from the [Edge Policy Console](../../about/edge-policy-console), this document covers various approaches and advanced use cases for debugging Ambassador issues.
+
+We assume that you already have a running Ambassador installation. Our examples use the following CRDs as the Ambassador configuration:
+
+?
+
+First, create an example configuration for debugging demonstrations that are available throughout this document.
+
+Then, complete the following debugging options:
+ 
+* Check Ambassador Status
+* Review Ambassador Logs
+* Examine Pod and Container Contents
+
+## Example Configuration
+
+To see a demonstration of how you can debug your Ambassador instance, follow the instructions to create an example mapping configuration.
+
+Note: The following assumes that you deployed Ambassador and the following services from the [quick start installation guide](../../user-guide/install) to a Kubernetes cluster.
+
+**In the command line:**
+
+1. Create a cluster in GKE with RBAC support enabled, and your user account configured correctly. Then run the following:
+
+    ```console
+    $ gcloud container clusters create ambassador-demo --preemptible
+    $ kubectl create clusterrolebinding cluster-admin-binding-new \
+    --clusterrole cluster-admin --user <your_user_name>
+    ```
+
+2. Deploy the latest version of Ambassador following the instructions from [here](../../user-guide/install).
+3. Next, create an Ambassador Service and deploy a basic `httpbin` Ambassador Mapping by saving the following YAML to a file named `ambassador-services.yaml`
+
+   ```yaml
+    ---
+    apiVersion: getambassador.io/v2
+    kind:  Mapping
+    metadata:
+      name:  httpbin
+    spec:
+      prefix: /httpbin/
+      service: httpbin.org
+      host_rewrite: httpbin.org
+    ```
+
+4. Apply this into your cluster with the following command:
+
+    ```console
+    $ kubectl apply -f ambassador-services.yaml
+    ```
+
+You should now be able to utilize this example mapping for debugging demonstrations and purposes.
+
+## Check Ambassador Status
+
+First, check to see if the [Diagnostics](../diagnostics) service is reachable. The following command assumes you have installed Ambassador in the `ambassador` name space: `kubectl port-forward -n ambassador_<pod name> 8877`
```

Comment:
The `kubectl` command really has this basic grammar:

```
kubectl verb  [-n namespace] object [object [...]]
```

where I'm omitting the additional switches that can be present for clarity. In Ambassador's case, the `namespace` will usually be `ambassador`, _and_ the `object` may also have a name starting with "ambassador", but _those "ambassador"s are not the same thing_. 

So for this `port-forward`, they should start with 

```
kubectl get pods -n ambassador
```

to get the list of pods in the `ambassador` namespace, which will be something like

```
NAME                         READY     STATUS    RESTARTS   AGE
ambassador-85c4cf67b-4pfj2   1/1       Running   0          1m
ambassador-85c4cf67b-fqp9g   1/1       Running   0          1m
ambassador-85c4cf67b-vg6p5   1/1       Running   0          1m
```

(pasted from below). Each bit like `ambassador-85c4cf67b-4pfj2` is a pod name, so they'd then run 

```
kubectl port-forward -n ambassador ambassador-85c4cf67b-4pfj2 8877
```

where, yup, the fact that you see `ambassador` repeated twice is not a bug.

In the past I documented this much as above, using a literal pod name, and then saying "Of course you'll need to paste in the actual name of one of your pods." Not that I've ever known how well that worked.